sale_commission/models/account_invoice.py

```diff
@@ -128,13 +123,13 @@ def onchange_agent(self):
     @api.depends('commission.commission_type', 'invoice_line.price_subtotal')
     def _get_amount(self):
         self.amount = 0.0
-        if (not self.invoice_line.product_id.commission_free and
-                self.commission):
-            subtotal = self.invoice_line.price_subtotal
-            if self.commission.commission_type == 'fixed':
-                self.amount = subtotal * (self.commission.fix_qty / 100.0)
-            else:
-                self.amount = self.commission.calculate_section(subtotal)
+        sign = {
+            'out_invoice': 1, 'in_invoice': -1,
+            'out_refund': -1, 'in_refund': 1,
+        }[self.invoice.type or 'out_invoice']
+        amount = self.commission.compute_invoice_commission(
```

Comment:
Making this, you have obscured what are the depending fields for the computation, so I recommend you not to do it (that's why I didn't do it). Imagine you include another field for the computation. Having decoupled the computation method, you won't remember to add that field to the depends section. And also you don't win anything, because this computation is invoice line specific.

What you can do is to refactor a compute commission method that gets amount as parameter (or any other you need in subsequent PRs), and return the corresponding commission